test/submit-test.js

```diff
@@ -0,0 +1,433 @@
+      .reply(200, "<html><body><form method='POST' action='http://www.expressen.se'><button type='submit'></button></form></body></html>", {
+        "content-type": "text/html"
+      });
+
+    let browser = await Browser(app, {
+      headers: {host: "www.expressen.se"}
+    }).navigateTo("/");
+
+    const form = browser.document.getElementById("multi-submit-form");
+    const button = form.getElementsByTagName("button")[2];
+
+    button.click();
+
+    browser = await browser._pending;
+
+    expect(browser.window.location.href).to.equal("https://www.example.com/1");
+  });
+
+  it("named submit element with value send payload with value", async () => {
+    nock("https://www.example.com")
+      .post("/2", (body) => {
+        expect(body).to.have.property("named-button-with-value").that.equal("1");
+        return true;
+      })
+      .matchHeader("host", "www.example.com")
+      .reply(200, "<html><body><form method='POST' action='http://www.expressen.se'><button type='submit'></button></form></body></html>", {
+        "content-type": "text/html"
+      });
+
+    let browser = await Browser(app, {
+      headers: {host: "www.expressen.se"}
+    }).navigateTo("/");
+
+    const form = browser.document.getElementById("multi-submit-form");
+    const button = form.getElementsByTagName("button")[3];
+
+    button.click();
+
+    browser = await browser._pending;
+
+    expect(browser.window.location.href).to.equal("https://www.example.com/2");
+  });
+
+  it("submits to local app if absolute form action matches host header", async () => {
+    let browser = await Browser(app, {
+      headers: {host: "www.expressen.se"}
+    }).navigateTo("/");
+
+    const form = browser.document.getElementById("post-form-absolute-url");
+    const button = form.getElementsByTagName("button")[0];
+
+    button.click();
+
+    expect(browser._pending).to.be.ok;
+
+    browser = await browser._pending;
+
+    expect(browser.window.location.href).to.equal("http://www.expressen.se/");
+  });
+
+  it("submits to local app if absolute form action matches x-forwarded-host header", async () => {
+    let browser = await Browser(app, {
+      headers: {
+        host: "some-other-host.com",
+        "x-forwarded-host": "www.expressen.se"
+      }
+    }).navigateTo("/");
+
+    const form = browser.document.getElementById("post-form-absolute-url");
+    const button = form.getElementsByTagName("button")[0];
+
+    button.click();
+
+    expect(browser._pending).to.be.ok;
+
+    browser = await browser._pending;
+
+    expect(browser.window.location.href).to.equal("http://www.expressen.se/");
+  });
+
+  it("form.submit() emits no submit event", async () => {
+    const browser = await Browser(port).navigateTo("/");
+
+    const form = browser.document.getElementById("post-form");
+
+    let submitEvent = false;
+    browser.document.addEventListener("submit", () => submitEvent = true);
+
+    form.submit();
+
+    expect(browser._pending).to.be.ok;
```

Comment:
this assures that form submission is happening, right?

Reply:
yes. The "browser" has unloaded and fetched the response from the form submission and returned a new browser with that content. We should consider some other name than `_pending`... a bit unclear.